pooch/tests/test_core.py

```diff
@@ -53,18 +55,22 @@ def test_pooch_custom_url():
         urls = {"tiny-data.txt": BASEURL + "tiny-data.txt"}
         # Setup a pooch in a temp dir
         pup = Pooch(path=path, base_url="", registry=REGISTRY, urls=urls)
-        # Check that the warning says that the file is being updated
-        with warnings.catch_warnings(record=True) as warn:
-            fname = pup.fetch("tiny-data.txt")
-            assert len(warn) == 1
-            assert issubclass(warn[-1].category, UserWarning)
-            assert str(warn[-1].message).split()[0] == "Downloading"
-            assert str(warn[-1].message).split()[-1] == "'{}'.".format(path)
+        # Create an in-memory file and a logging handler that writes events to
+        # this file, and check that downloading data gets logged
+        log_file = io.StringIO()
+        handler = logging.StreamHandler(log_file)
+        get_logger().addHandler(handler)
+        fname = pup.fetch("tiny-data.txt")
+        logs = log_file.getvalue()
+        assert logs.split()[0] == "Downloading"
+        assert logs.split()[-1] == "'{}'.".format(path)
```

Comment:
This would be great as a context manager:

```python
with capture_log() as log:
    fname = pup.fetch("tiny-data.txt")
    logs = log.getvalue()
    assert logs.split()[0] == "Downloading"
    assert logs.split()[-1] == "'{}'.".format(path)
```

The context would setup the handler and return the `StringIO`. When exiting it could do any clean up that it needs (remove the handler so it won't affect other tests down the line?). 

An easy way make a context manager is with [`contextlib`](https://docs.python.org/3.8/library/contextlib.html):

```python
from contextlib import contextmanager

@contextmanager
def capture_log():        
    log_file = io.StringIO()
    handler = logging.StreamHandler(log_file)
    get_logger().addHandler(handler)
    # return this to the context and halt until exit
    yield log_file
    # executed on exit
    get_logger().removeHandler(handler)
```

which could be added to `pooch/tests/utils.py`.

Reply:
Ooh this is a big improvement!

Reply:
:+1: Looks great!